internal/extender/resourcereservations.go

```diff
@@ -335,6 +336,14 @@ func (rrm *ResourceReservationManager) bindExecutorToResourceReservation(ctx con
 	if err != nil {
 		return werror.WrapWithContextParams(ctx, err, "failed to update resource reservationName", werror.SafeParam("reservationName", reservationName))
 	}
+
+	// only report the metric the first time the reservation is bound
+	if _, ok := resourceReservation.Status.Pods[reservationName]; !ok {
```

Comment:
I have convinced myself this is the check we want. We don't care why a pod was bound to a reservation, and yet we're still in this method - it would be quite complex to handle all cases. Let's start with first bound time, but please update the metric to name to indicate that.

https://github.com/palantir/k8s-spark-scheduler/blob/2112d580b54a922c5ca5fd7cc4723c83a4272201/internal/extender/resourcereservations.go#L361-L382